content/posts/2021/reana-client-python-api.md

```diff
@@ -0,0 +1,223 @@
+            'commands': [
+                'root -b -q \'code/fitdata.C("${data}","${plot}")\' | tee fitdata.log'
+            ],
+        },
+    ]
+}
+```
+
+### Workflow creation and usage
+
+Finally, once all the workflow steps are specified, we are ready to create it
+in REANA. To create the workflow using `reana-client` we could use [create](https://reana-client.readthedocs.io/en/latest/#reana-client-create)
+command. In our case we can use `create_workflow_from_json` function
+where we need to define all the needed parameters:
+
+```python
+from reana_client.api.client import create_workflow_from_json
+
+create_workflow_from_json(
+    workflow_json=my_workflow,
+    name=my_workflow_name,
+    access_token=my_reana_token,
+    parameters=my_inputs,
+    workflow_engine=my_workflow_type)
+```
+
+After creating our workflow we also need to upload all the input files to
+the workflow workspace. Same functionality can be achieved with `reana-client`
+[upload](https://reana-client.readthedocs.io/en/latest/#reana-client-upload)
+command. It can be done as follows:
+
+```python
+from reana_client.api.client import upload_to_server
+
+abs_path_to_input_files = [os.path.abspath(f) for f in my_inputs['files']]
+upload_to_server(my_workflow_name, abs_path_to_input_files, my_reana_token)
+```
+
+At this point we have done all what is needed to start our workflow in REANA.
+To do it with `reana-client` we could call [start](https://reana-client.readthedocs.io/en/latest/#reana-client-start)
+command. But in this case it can be started by simply calling `start_workflow`
+function and defining workflow name and access token:
+
+```python
+from reana_client.api.client import start_workflow
+
+start_workflow(my_workflow_name, my_reana_token, {})
+```
+
+After starting our wokflow we might want to check what is the current status of it.
+To inspect the workflow status using `reana-client` we could use [status](https://reana-client.readthedocs.io/en/latest/#reana-client-status)
+command. Similarly, it can be achieved with `get_workflow_status` function.
+Since it will take couple of seconds for all the computations to be finished
+while executing our workflow, we will use a loop to wait and regurlary check
+what is the current status of our workflow:
+
+```python
+import time
+
+while True:
+    from reana_client.api.client import get_workflow_status
+
+    status_details = get_workflow_status(my_workflow_name, my_reana_token)
+    if status_details['status'] == 'finished':
+        break
+    print('Current status: ', status_details['status'])
+    time.sleep(1)
+```
+
+Once the workflow is finished, we usually want to list all the output files which our workflow
+produced. With `reana-client` it can be achieved by using
+[ls](https://reana-client.readthedocs.io/en/latest/#reana-client-ls)
+command. Similarly, it can be done in the following manner:
+
+```python
+from reana_client.api.client import list_files
+
+list_files(my_workflow_name, my_reana_token)
```

Comment:
Meta-comment: It may be useful to also show outputs of each command, in a notebook-like style... In this way, people reading the blog post will be able to picture fully the interactions, what is returned, without actually even running it.